xlators/protocol/client/src/client-rpc-fops.c

```diff
@@ -5100,6 +5129,13 @@ client3_3_lk(call_frame_t *frame, xlator_t *this, void *data)
             client_is_setlk(local->cmd)) {
             client_add_lock_for_recovery(local->fd, args->flock, &local->owner,
                                          local->cmd);
+        } else if (local->check_reopen) {
```

Comment:
Even when op_errno is say ENOTCONN, is_reopen_allowed will be set with FD_BAD. Is that expected?

Reply:
Yes, it is expected. This is used only to identify whether client_pre_lk(_v2) is successful or not. If it fails for any reason ENOTCONN/EBADFD, it will send back afr FD_BAD if strict-locks is enabled. For afr this just means that the getlk on that brick is failed for some reason. We do not use this as op_errno.